app/models/concerns/hostext/search.rb

```diff
@@ -171,6 +171,12 @@ def search_by_hostgroup_and_descendants(key, operator, value)
 
       def search_by_params(key, operator, value)
         key_name = key.sub(/^.*\./, '')
+
+        if Parameter.find_by(name: key_name)&.key_type == 'boolean'
```

Comment:
I've been thinking about a solution where we don't have to call the database to get the parameter type, but so far, I've had no luck.